core/scss/components/site-search/_site-search.scss

```diff
@@ -23,10 +27,6 @@
     color: $color-black;
     opacity: 1; // For Firefox
   }
-
-  @media #{$breakpoint-lg} {
```

Comment:
No max width by default. Leave it up to the user to put it in a container of the appropriate size.